libs/core/src/lib/utils/directives/truncation/truncation.directive.ts

```diff
@@ -0,0 +1,55 @@
+import { AfterViewInit, Directive, ElementRef, Input } from '@angular/core';
+
+@Directive({
+    // eslint-disable-next-line @angular-eslint/directive-selector
+    selector: '[fd-truncate]'
+})
+export class TruncationDirective implements AfterViewInit {
+    @Input() charLength: number;
+    @Input() pixLength: number;
+
+    /** @hidden */
+    private defaultTruncationByChar = 30;
+
+    constructor(private _elementRef: ElementRef) {}
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        const rootElement = this._elementRef.nativeElement;
+        rootElement.title = rootElement.textContent; // adds title to element to enable mouse over tooltip
+
+        if (this.pixLength) {
+            this._truncateByPixel(rootElement);
+        } else {
+            this._truncateByChar(rootElement);
+        }
+    }
+
+    /** @hidden
+     * Truncates element by width
+     */
+    private _truncateByPixel(_element): void {
```

Comment:
```suggestion
    private _truncateByPixel(_element: HTMLElement): void {
```

Nice code! Let's do types for function parameters